pinot-segment-local/src/main/java/org/apache/pinot/segment/local/function/InternalReduceFunctions.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.segment.local.function;
+
+import org.apache.pinot.segment.local.customobject.PinotFourthMoment;
+import org.apache.pinot.spi.annotations.ScalarFunction;
+
+
+/**
+ * This class contains functions that are necessary for the multistage engine
+ * aggregations that need to be reduced after the initial aggregation to get
+ *
+ */
+public class InternalReduceFunctions {
```

Comment:
In the existing engine code, the custom objects are defined in `pinot-segment-local` because they are created / serialized on the leaf layer before sending. 

But I don't think reduce functions should be defined in `pinot-segment-local`. This is not going to run on leaf layer right ? 

Putting this in `pinot-segment-local` does not seem intuitive imo.

Reply:
Or may be I misunderstood how you have separated them into aggregate and reduce. If this is going to run on the leaf, then package structure is fine.

Reply:
It relies on the `PinotFourthMoment` class, which is in `pinot-segment-local` because that's what the aggregate uses internally and that's where all the other custom objects are. I can move it to any package that depends on `pinot-segment-local` if you have any suggestions of where's a better place.

Reply:
putting it in pinot-core makes more sense to me

Reply:
moved to `pinot-core`

Reply:
sgtm